### PR TITLE
Fix some python warnings

### PR DESCRIPTION
**Pull request by electretmike**
I got some warning when calling the python scripts. This changes fixes those warnings.

### CODE REVIEW

File: src/hid/parser.py [hunk 1]
```diff
@@ -12,10 +12,11 @@
...
     """Failed to parse a line in file"""
-    def __init__(self, filepath, line):
-        super().__init__('HID usage parsing error in file ' + filename + ' (line="' + line + '")')
-    def __init__(self, page):
-        super().__init__('HID usage parsing error in page ' + page.name())
+    def __init__(self, filepath=None, line=None, page=None):
+        if (filepath is not None):
+            super().__init__('HID usage parsing error in file ' + filepath + ' (line="' + line + '")')
+        else:
+            super().__init__('HID usage parsing error in page ' + page.name())
```
Comment (benedekkupper) [inline src/hid/parser.py:19]: Could you please leave this constructor intact, and create an additional one that only takes the page parameter instead?

**Comment by benedekkupper** [comment]
Hi! I got the time to check this in detail, I went ahead and added some further improvements in addition, please let me know if any further issues are present with the latest version.

What is your use-case for this project?

**Comment by electretmike** [comment]
I was going to use this to generate descriptors for a USB device. But it turned out that integrating python in out build-system was tricky, so had to switch to a c++ library.

**Comment by benedekkupper** [comment]
Well you could also just generate the code once, and commit it to git, these tables barely see any change these last years.

**Pull request closed without merging**